sphinx/builders/linkcheck.py

```diff
@@ -717,8 +675,8 @@ def setup(app: Sphinx) -> ExtensionMetadata:
     app.add_config_value('linkcheck_anchors_ignore', ['^!'], '')
     app.add_config_value('linkcheck_anchors_ignore_for_url', (), '', (tuple, list))
     app.add_config_value('linkcheck_rate_limit_timeout', 300.0, '', (int, float))
-    app.add_config_value('linkcheck_allow_unauthorized', True, '')
-    app.add_config_value('linkcheck_report_timeouts_as_broken', True, '', bool)
+    app.add_config_value('linkcheck_allow_unauthorized', False, '', bool)
```

Comment:
```suggestion
    app.add_config_value('linkcheck_allow_unauthorized', False, '')
```

I had snuck in a type definition change here, but let's follow best-practice and not do that with unrelated changes.